### PR TITLE
DF-3634 ldap update to allow users to be added to the "Users" container

### PR DESCRIPTION
**Pull request by mhofert-r7**
### Description

Describe the proposed changes:

  - Update add user action to allow users to be added to the Users container

### Testing

- See testing output in comment below

## PR Requirements

Developers, verify you have completed the following items by checking them off:

### Style

Review the [style guide](https://komand.github.io/python/style.html)

- [ ] For dependencies, pin [OS package](https://komand.github.io/python/style.html#dockerfile) and [Python package](https://komand.github.io/python/style.html#requirements-txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://komand.github.io/python/sdk.html#sdk-versions) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [ ] For error handling, use of [PluginException](https://komand.github.io/python/error_handling.html#plugin-exceptions) and [ConnectionTestException](https://komand.github.io/python/error_handling.html#connection-exceptions)
- [ ] For logging, use [self.logger](https://komand.github.io/python/sdk.html#logging)
- [ ] For docs, use [changelog style](https://komand.github.io/python/style.html#changelog)
- [ ] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://komand.github.io/python/style.html#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `make validate` and make sure everything passes
2. Run the assessment tool: `icon-plugin run -A -R all -T all`. For single action validation: `icon-plugin run -A -R tests/my_action.json -T tests/my_action.json`
3. Copy (`icon-plugin ... | pbcopy`) and paste the output in **a new post** on this PR.
4. Add UI screenshot of the workflow used for testing
5. Add UI screenshot of the job output used for testing
6. Add UI screenshot of the artifact (See rules in UI Checklist) used for testing

### CODE REVIEW

**Comment by mhofert-r7** [comment]
## Assessment
### Run

<details>

```
{
  "body": {
    "log": "Connecting to 192.168.1.10:636\nConnected!\nrapid7/Active Directory LDAP:3.2.8. Step name: add_user\nUser DN=CN=Test Person,CN=Users,DC=mattsdomain,DC=local\n{'givenName': 'Test', 'sn': 'Person', 'sAMAccountName': 'tperson', 'userPassword': 'testtest04!', 'userPrincipalName': 'tperson@mattsdomain.local'}\n{'result': 0, 'description': 'success', 'dn': '', 'message': '', 'referrals': None, 'type': 'modifyResponse'}\n",
    "meta": {},
    "output": {
      "success": true
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/active_directory_ldap:3.2.8 run < tests/add_user.json
</summary>
</details>

### UI

Screenshots of the plugin being used in Komand for validating use go below this line.

#### Workflow Builder


#### Job


#### Artifact

File: active_directory_ldap/tests/add_user.json [hunk 1]
```diff
@@ -2,13 +2,6 @@
     "body": {
         "action": "add_user",
         "connection": {
-            "host": "",
```
Comment (jschipp-r7) [inline active_directory_ldap/tests/add_user.json]: We should keep all these keys around so devs can easily try out the plugin by entering input. Same for all the JSON test files.

File: active_directory_ldap/tests/disable_user.json [hunk 2]
```diff
@@ -2,16 +2,9 @@
...
-            "port": "",
-            "use_ssl": "",
-            "username_password": {
-                "password": "",
-                "username": ""
-            }
         },
         "input": {
-            "distinguished_name": "CN=hollis, brown,OU=domain_users,DC=mattsdomain,DC=local"
+            "distinguished_name": ""
```
Comment (jschipp-r7) [inline active_directory_ldap/tests/disable_user.json]: Sample input is also good to have for the distinguished name

File: active_directory_ldap/tests/query.json [hunk 3]
```diff
@@ -2,17 +2,10 @@
...
-            "port": "",
-            "use_ssl": "",
-            "username_password": {
-                "password": "",
-                "username": ""
-            }
         },
         "input": {
-            "search_base": "DC=mattsdomain,DC=local",
-            "search_filter": "(distinguishedName=CN=hollis, brown,OU=domain_users,DC=mattsdomain,DC=local)"
```
Comment (jschipp-r7) [inline active_directory_ldap/tests/query.json]: Sample input is also good to have for the search filter

File: active_directory_ldap/tests/add_user.json (hunk 1 above)
Comment (mhofert-r7) [reply active_directory_ldap/tests/add_user.json]: ok it's like I thought having these keys will cause us to fail validation.

Comment (jschipp-r7) [reply active_directory_ldap/tests/add_user.json]: It should pass validation having empty strings. I tested this with another plugin..

@mhofert-r7 But I do see that a number of the JSON test files are not valid JSON in this plugin.

```
[*] Executing validator JSONValidator
Validator JSONValidator failed!
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/icon_validator/validate.py", line 29, in validate
    v.validate(spec)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/icon_validator/rules/json_validator.py", line 20, in validate
    raise Exception(f"The following test files are not in proper JSON format: {JSONValidator.invalid_files}")
Exception: The following test files are not in proper JSON format: ['delete.json', 'move_object.json', 'modify_groups.json', 'enable_user.json', 'query.json', 'disable_user.json', 'force_password_reset.json', 'reset_password.json']
```

Example test
```
$ jq '.' tests/disable_user.json
parse error: Unfinished JSON term at EOF at line 19, column 1
```

Comment (jschipp-r7) [reply active_directory_ldap/tests/add_user.json]: We'll need to fix these.

File: active_directory_ldap/tests/add_user.json [hunk 4]
```diff
@@ -3,23 +3,23 @@
...
         "connection": {
             "host": "",
-            "port": "",
-            "use_ssl": "",
+            "port": 636,
+            "use_ssl": true,
             "username_password": {
-                "password": "",
-                "username": ""
+                "password": null,
```
Comment (jschipp-r7) [inline active_directory_ldap/tests/add_user.json]: We'll need to revert this back, per request in earlier review https://github.com/rapid7/insightconnect-plugins/pull/230#discussion_r360510374

File: active_directory_ldap/plugin.spec.yaml [hunk 5]
```diff
@@ -3,8 +3,8 @@ extension: plugin
 products: [insightconnect]
 name: active_directory_ldap
 title: Active Directory LDAP
-description: Interfaces with Microsofts Active Directory service to manage domains, users, and objects within a network
-version: 3.2.7
+description: This plugin utilizes Microsoft\'s Active Directory service to create and manage domains, users, and objects within a network
```
Comment (mrinehart-r7) [inline active_directory_ldap/plugin.spec.yaml]:
```suggestion
description: Interface with the Microsoft Active Directory service to create and manage domains, users, and objects within a network
```

**Pull request merged**